Sources/Consent.swift

```diff
@@ -79,12 +86,17 @@ public class Consent: NSObject, Extension {
             return
         }
 
-        // set metadata
-        newPreferences.setTimestamp(date: event.timestamp)
-        preferencesManager.mergeAndUpdate(with: newPreferences)
-        shareCurrentConsents(event: event)
-        // Share only changed preferences instead of all preferences to prevent accidental sharing of default consents.
-        dispatchEdgeConsentUpdateEvent(preferences: newPreferences)
+        // Only proceed with Edge event dispatch if preferences actually changed
+        if preferencesManager.mergeAndUpdate(with: newPreferences) || shouldForceSync(event: event) {
+            // Add timestamp after checking for change in preferences to prevent false positive from timestamp differences.
+            newPreferences.setTimestamp(date: event.timestamp)
+            preferencesManager.mergeAndUpdate(with: newPreferences) // re-apply with updated metadata
```

Comment:
why do we need to reapply the new preferences?

Reply:
The ConsentPreferences class checks for equality using the entire underlying dictionary, including the timestamp. The first merge is used to check the equality of the current and new (without timestamp) preferences. If they are not equal, then the event timestamp is added to the new preferences and they are merged again to update the timestamp.